ejb3/src/main/java/org/jboss/as/ejb3/security/RunAsPrincipalInterceptor.java

```diff
@@ -35,16 +39,30 @@
 import org.wildfly.security.auth.server.SecurityDomain;
 import org.wildfly.security.auth.server.SecurityIdentity;
 import org.wildfly.security.authz.AuthorizationFailureException;
+import org.wildfly.security.manager.WildFlySecurityManager;
 
 /**
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public class RunAsPrincipalInterceptor implements Interceptor {
     public static final String ANONYMOUS_PRINCIPAL = "anonymous";
+
     private final String runAsPrincipal;
+    private final boolean securityRequired;
 
-    public RunAsPrincipalInterceptor(final String runAsPrincipal) {
+    public RunAsPrincipalInterceptor(final String runAsPrincipal, boolean securityRequired) {
         this.runAsPrincipal = runAsPrincipal;
+        this.securityRequired = securityRequired;
+    }
+
+    private boolean remotingContextIsSet (){
+        RemoteConnection remoteConnection;
+        if(WildFlySecurityManager.isChecking()) {
+            remoteConnection = WildFlySecurityManager.doUnchecked((PrivilegedAction<RemoteConnection>) () -> RemotingContext.getRemoteConnection());
+        } else {
+            remoteConnection = RemotingContext.getRemoteConnection();
```

Comment:
The `RemotingContext` class is meant to be used for legacy security, I don't think we should be using it for Elytron.

Reply:
Ok it might not be needed

Reply:
@fjuma Ok it might not be needed